ls1intum/Artemis
# `General`: Merge scores and submission pages into one page for exercise instructors

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


The submission and scores tables of each exercise nearly showed the same information but created many duplications.

### Description

This PR merges these to tables in the following way:
- The submission table and the buttons redirecting to it are removed. If any place needs a redirect to the scores table and does not already have it, let me know.
- The scores table always shows all participations with a submission and in case of a submission without result displays "no result" instead.
- The filter now have more options and are moved into one drop down instead of multiple options next to each other
- The buttons redirecting to the manual assessment (and code editor) are now consistently shown in each exercise and configuration as long as manual assessments or editing code is possible.
- Admins can no longer go to the user editing page from the scores table.
- The assessment overview does not have a link to the own assessed submissions, but all assessments are still visible in the assessment overview

### Steps for Testing

Prerequisites:
- 1 Instructor
- 3 Students
- 1 Exercise of each type

1. Log in to Artemis
2. Participate with each student in the exercises, on only starts the exercise, one only submits something, one also gets an assessment
3. Check the scores page and all related views like the exercise overview or the assessment view

#### Exam Mode Testing


Same as normal mode, but with the exercises in an exam

### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| exercise-scores.component.ts | 75% | 90% |
| exercise.utils.ts | 90% | 88% |

### Screenshots


#### Programming exercises (automatic but with complaints):
Old scores table:
![Bildschirm­foto 2023-03-14 um 21 02 38](https://user-images.githubusercontent.com/38322605/225124330-fb7e298b-6754-4ae7-8b3e-e2128f684a68.png)
Old submissions table:
![Bildschirm­foto 2023-03-14 um 21 02 45](https://user-images.githubusercontent.com/38322605/225124355-80c857be-0912-4777-96ef-cd78d6c2d924.png)
New merged table:
![Bildschirm­foto 2023-03-15 um 11 03 16](https://user-images.githubusercontent.com/38322605/225277220-44341d85-b2c2-45cb-94ae-89eaea6fe79c.png)
#### Modeling/Text/Fileupload exercises:
Old scores table:
![Bildschirm­foto 2023-03-15 um 09 28 18](https://user-images.githubusercontent.com/38322605/225251363-ff2ee84b-38d6-4fe2-bfbf-8ee9cf780ada.png)
Old submissions table:
![Bildschirm­foto 2023-03-15 um 09 28 32](https://user-images.githubusercontent.com/38322605/225251387-a9b32273-ad72-4443-b9d9-0fd4fd47e855.png)
New merged table:
![Bildschirm­foto 2023-03-15 um 09 24 35](https://user-images.githubusercontent.com/38322605/225251344-291b1fcd-5167-41e4-b207-59fe82d154a7.png)
#### Quiz exercises:
Did not have a submission table.
New merged table:
![Bildschirm­foto 2023-03-15 um 11 15 22](https://user-images.githubusercontent.com/38322605/225278953-ccda83bd-d3d9-41eb-8244-a54ef5d644d7.png)
#### Filter and export:
![Bildschirm­foto 2023-03-15 um 11 03 23](https://user-images.githubusercontent.com/38322605/225277344-42af6957-198e-4ecb-961d-89940f2c9fb8.png)
![Bildschirm­foto 2023-03-15 um 11 06 25](https://user-images.githubusercontent.com/38322605/225277368-8be0e7f7-8604-4ef5-b47d-81b03b45ebe9.png)
